File: src/AI.Forged.TourOps.Domain/Entities/Rate.cs
using AI.Forged.TourOps.Domain.Enums;

namespace AI.Forged.TourOps.Domain.Entities;

public class Rate
{
    public Guid Id { get; set; }
    public Guid ProductId { get; set; }
    public DateOnly SeasonStart { get; set; }
    public DateOnly SeasonEnd { get; set; }
    public PricingModel PricingModel { get; set; }
    public decimal BaseCost { get; set; }
    public string Currency { get; set; } = string.Empty;
    public int? MinPax { get; set; }
    public int? MaxPax { get; set; }
    public decimal? ChildDiscount { get; set; }
    public decimal? SingleSupplement { get; set; }
    public int? Capacity { get; set; }
    public DateTime CreatedAt { get; set; }

    public Product Product { get; set; } = null!;
}

File: ui/src/types/rate.ts
export type PricingModel = 'PerPerson' | 'PerGroup' | 'PerUnit';

export interface Rate {
  id: string;
  productId: string;
  productRoomId?: string | null;
  productRoomName?: string | null;
  isActive: boolean;
  previousRateId?: string | null;
  supersededAt?: string | null;
  seasonStart: string;
  seasonEnd: string;
  pricingModel: PricingModel;
  baseCost: number;
  currency: string;
  minPax?: number | null;
  maxPax?: number | null;
  childDiscount?: number | null;
  singleSupplement?: number | null;
  capacity?: number | null;
  validityPeriod?: string | null;
  validityPeriodDescription?: string | null;
  rateVariation?: string | null;
  rateTypeName?: string | null;
  rateBasis?: string | null;
  occupancyType?: string | null;
  mealBasis?: string | null;
  minimumStay?: string | null;
  createdAt: string;
}

export interface RateRequest {
  productId: string;
  productRoomId?: string | null;
  seasonStart: string;
  seasonEnd: string;
  pricingModel: PricingModel;
  baseCost: number;
  currency: string;
  minPax?: number | null;
  maxPax?: number | null;
  childDiscount?: number | null;
  singleSupplement?: number | null;
  capacity?: number | null;
  validityPeriod?: string | null;
  validityPeriodDescription?: string | null;
  rateVariation?: string | null;
  rateTypeName?: string | null;
  rateBasis?: string | null;
  occupancyType?: string | null;
  mealBasis?: string | null;
  minimumStay?: string | null;
}
